tobago-example/tobago-example-demo/src/main/webapp/content/30-concept/53-collapsible/00-collapsible-box/Collapsible_Box.test.js

```diff
@@ -15,306 +15,206 @@
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.ok(content1Fn() !== null);
-    assert.equal(content2Fn().length, content2Length);
-  });
-  TTT.action(function () {
-    show2Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.ok(content1Fn() !== null);
-    assert.ok(content2Fn() !== null);
-  });
-  TTT.startTest();
+  let test = new JasmineTestTool(done);
+  test.do(() => show1Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() && content1Fn() !== null);
+  test.do(() => expect(content1Fn() !== null).toBe(true));
+  test.do(() => expect(content2Fn().length).toBe(content2Length));
+  test.do(() => show2Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() && content1Fn() !== null);
+  test.do(() => expect(content1Fn() !== null).toBe(true));
+  test.do(() => expect(content2Fn() !== null).toBe(true));
+  test.start();
 });
 
-QUnit.test("Full Server Request: open box 1, close box 2", function (assert) {
-  let show1Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:show1");
-  let hide2Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:hide2");
-  let content1Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:content1");
-  let content2Fn = testFrameQuerySelectorFn("#page\\:mainForm\\:server\\:content2");
+it("Full Server Request: open box 1, close box 2", function (done) {
+  let show1Fn = querySelectorFn("#page\\:mainForm\\:server\\:show1");
+  let hide2Fn = querySelectorFn("#page\\:mainForm\\:server\\:hide2");
+  let content1Fn = querySelectorFn("#page\\:mainForm\\:server\\:content1");
+  let content2Fn = querySelectorFn("#page\\:mainForm\\:server\\:content2");
   let content2Length = content2Fn().length;
 
-  let TTT = new TobagoTestTool(assert);
-  TTT.action(function () {
-    show1Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.ok(content1Fn() !== null);
-    assert.equal(content2Fn().length, content2Length);
-  });
-  TTT.action(function () {
-    hide2Fn().dispatchEvent(new Event("click", {bubbles: true}));
-  });
-  TTT.waitForResponse();
-  TTT.asserts(2, function () {
-    assert.ok(content1Fn() !== null);
-    assert.equal(content2Fn(), null);
-  });
-  TTT.startTest();
+  let test = new JasmineTestTool(done);
+  test.do(() => show1Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() && content1Fn() !== null);
+  test.do(() => expect(content1Fn !== null).toBe(true));
+  test.do(() => expect(content2Fn().length).toBe(content2Length));
+  test.do(() => hide2Fn().dispatchEvent(new Event("click", {bubbles: true})));
+  test.wait(() => content1Fn() && content1Fn() !== null);
+  test.do(() => expect(content1Fn !== null).toBe(true));
+  test.do(() => expect(content2Fn()).toBe(null));
```

Comment:
For cleaner code you might use the same 'expect' as in line 84 and write: test.do(() => expect(content2Fn() !== null).toBe(false));